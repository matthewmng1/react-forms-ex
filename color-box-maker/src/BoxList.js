import React, {useState} from "react";
import Box from "./Box"
import {v4 as uuid} from "uuid"
import NewBoxForm from "./NewBoxForm";

const BoxList = () => {
  const INITIAL_STATE = []

  const [box, setBox] = useState(INITIAL_STATE)
  const addBox = (newBox) => {
    setBox(box => [...box, newBox])
  }
  const removeBox = (id) => {
    setBox(box => box.filter(box => box.id !== id))
  }

  return(
    <div>
      <h1>Color Box Maker</h1>
      <NewBoxForm addBox ={addBox}/>
      <div>
        {box.map(({id, width, height, color}) => 
        <Box id={id} width={width} height={height} color={color} handleRemove={removeBox}/>)}
      </div>
    </div>
  )

}

export default BoxList;