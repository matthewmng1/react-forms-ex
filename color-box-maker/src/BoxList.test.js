import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

const addBox = (boxList, width="5", height="5", color="blue") => {
  const boxWidth = boxList.getByLabelText("Width:")
  const boxHeight = boxList.getByLabelText("Height:")
  const boxColor = boxList.getByLabelText("Color:")
  fireEvent.change(boxWidth, { target: { value: width } });
  fireEvent.change(boxHeight, { target: { value: height } });
  fireEvent.change(boxColor, { target: { value: color } });

  const btn = boxList.getByText("Add Color Box")
  fireEvent.click(btn);
}

it(`renders without crashing`, () => {
  render(<BoxList />);
});

it(`matches snapshop`, () => {
  const {asFragment} = render(<BoxList /> );
  expect(asFragment()).toMatchSnapshot();
})

it(`adds a new box to the page`, () => {
  const boxList = render(<BoxList />);

  expect(boxList.queryByText("Remove Box")).not.toBeInTheDocument();


  addBox(boxList);

  const rmvBtn = boxList.getByText("Remove Box");
  expect(rmvBtn).toBeInTheDocument();
  expect(rmvBtn.previousSibling).toHaveStyle(`
    width: 5em;
    height: 5em;
    background-color: blue`)
})