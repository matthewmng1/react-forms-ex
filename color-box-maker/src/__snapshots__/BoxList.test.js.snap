// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshop 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Color Box Maker
    </h1>
    <form>
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        placeholder="Enter Width"
        type="number"
        value=""
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        placeholder="Enter Height"
        type="number"
        value=""
      />
      <label
        for="color"
      >
        Color:
      </label>
      <input
        id="color"
        name="color"
        placeholder="Enter Color"
        type="text"
        value=""
      />
      <button>
        Add Color Box
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
